File: appmy/src/main/java/com/leilei/guoshujinfu/mylearning/util/ActivityCollector.java
package com.leilei.guoshujinfu.mylearning.util;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Administrator on 2017/8/3.
 */

public final class ActivityCollector {
    private static List<BaseActivity> activities = new ArrayList<>();
    public static void add(BaseActivity activity) {
        if(activity == null) {
            return;
        }
        activities.add(activity);
    }
    public static void remove(BaseActivity activity) {
        if(activity == null) {
            return;
        }
        activities.remove(activity);
    }
    public static void removeAll() {
        for(BaseActivity activity : activities){
            if(activity != null && !activity.isFinishing()) {
                activity.finish();
            }
            activities.clear();

        }
    }
}

File: yueting/src/main/kotlin/com/aqrlei/graduation/yueting/baselib/util/ActivityCollector.kt
package com.aqrlei.graduation.yueting.baselib.util

import com.aqrlei.graduation.yueting.baselib.base.BaseActivity

/**
 * @Author: AqrLei
 *@Name MyLearning
 *@Description:
 *@Date: 2017/8/22
 */
object ActivityCollector {
    private var activities = ArrayList<BaseActivity>()

    fun add(activity: BaseActivity?) {
        if (activity == null) return
        activities.add(activity)
    }

    fun remove(activity: BaseActivity?) {
        /*if(activity == null) return*/
        activities.remove(activity)
    }

    fun removeAll() {
        for (activity in activities ){
            if (activity != null && !activity.isFinishing) {
                activity.finish()
            }
            activities.clear()
        }
    }
}